src/app.py

```diff
@@ -47,7 +57,41 @@ def _log_filename() -> str:
 setup_logger(log_file=webhook_logs_dir / _log_filename())
 
 
+def signature_validator(func: Callable[[], tuple[str, int]]) -> Callable[[], tuple[str, int]]:
+    """Validate the signature of the incoming request.
+
+    Args:
+        func: function to be decorated.
+
+    Returns:
+        Decorated function.
+    """
+
+    @functools.wraps(func)
+    def wrapper() -> tuple[str, int]:
+        """Validate the signature of the incoming request.
+
+        Returns:
+            A tuple containing an error message and 403 status code if the signature is invalid.
+            Otherwise, the result of the decorated function.
+        """
+        if secret := app.config.get("WEBHOOK_SECRET"):
+            request_signature = request.headers.get(WEBHOOK_SIGNATURE_HEADER)
+            if request_signature is None:
+                return "X-Hub-signature-256 header is missing!", 403
+            try:
+                verify_signature(
+                    payload=request.data, secret_token=secret, signature_header=request_signature
+                )
+            except SignatureValidationError:
+                return "Signature validation failed!", 403
```

Comment:
Usually validation functions return booleans since it isn't exceptional for the validation function to encounter cases where the validation fails. Raising exceptions for invalid requests is exceptional if the function requires that the request is valid for it to work

Reply:
Updated